modules/ppcp-googlepay/src/Endpoint/UpdatePaymentDataEndpoint.php

```diff
@@ -0,0 +1,227 @@
+<?php
+/**
+ * Endpoint to update payment data like shipping method and address.
+ *
+ * @package WooCommerce\PayPalCommerce\Button\Endpoint
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\Googlepay\Endpoint;
+
+use Psr\Log\LoggerInterface;
+use Throwable;
+use WooCommerce\PayPalCommerce\ApiClient\Entity\Money;
+use WooCommerce\PayPalCommerce\Button\Endpoint\RequestData;
+use WooCommerce\PayPalCommerce\Button\Exception\RuntimeException;
+
+/**
+ * Class UpdatePaymentDataEndpoint
+ */
+class UpdatePaymentDataEndpoint {
+
+	const ENDPOINT = 'ppc-googlepay-update-payment-data';
+
+	/**
+	 * The request data helper.
+	 *
+	 * @var RequestData
+	 */
+	private $request_data;
+
+	/**
+	 * The logger.
+	 *
+	 * @var LoggerInterface
+	 */
+	private $logger;
+
+	/**
+	 * UpdatePaymentDataEndpoint constructor.
+	 *
+	 * @param RequestData     $request_data The request data helper.
+	 * @param LoggerInterface $logger The logger.
+	 */
+	public function __construct(
+		RequestData $request_data,
+		LoggerInterface $logger
+	) {
+		$this->request_data = $request_data;
+		$this->logger       = $logger;
+	}
+
+	/**
+	 * Returns the nonce.
+	 *
+	 * @return string
+	 */
+	public static function nonce(): string {
+		return self::ENDPOINT;
+	}
+
+	/**
+	 * Handles the request.
+	 *
+	 * @return bool
+	 * @throws RuntimeException When a validation fails.
+	 */
+	public function handle_request(): bool {
+		try {
+			$data = $this->request_data->read_request( $this->nonce() );
+
+			// Validate nonce.
+			if (
+				! isset( $data['nonce'] )
+				|| ! wp_verify_nonce( $data['nonce'], self::nonce() )
+			) {
+				throw new RuntimeException(
+					__( 'Could not validate nonce.', 'woocommerce-paypal-payments' )
+				);
+			}
```

Comment:
Why validate nonce here again? it's already validated here: https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-button/src/Endpoint/RequestData.php#L45-L53

Reply:
You're right. Added `$this->request_data->read_request( $this->nonce() );` after the nonce validation and didn't notice it was already validating it. Removed the duplicate.